batchflow/config.py

```diff
@@ -1,250 +1,257 @@
+            Parameters from kwargs also are parsed and saved to self.
         """
         if config is None:
-            self.config = Config.IAddDict()
-        elif isinstance(config, (dict, list)):
-            self.config = self.parse(config)
+            pass
         elif isinstance(config, Config):
-            self.config = config.config
+            super().__init__(config)
+        elif isinstance(config, (dict, list, tuple)):
+            self.parse(config)
         else:
-            raise TypeError(f'config must be dict, Config or list but {type(config)} was given')
+            raise TypeError(f'Config must be dict, Config, list or tuple but {type(config)} was given')
 
         for key, value in kwargs.items():
             self.put(key, value)
 
-    def pop(self, variables, config=None, **kwargs):
-        """ Returns variables and remove them from config
+    def parse(self, config):
+        """ Parses flatten config with slashes.
 
         Parameters
         ----------
-        variables : str or list of strs
-            names of variables. '/' is used to get value from nested dict
-        config : dict, Config or None
-            if None, variables will be getted from self.config else from config
+        config : dict, Config, list or tuple
 
         Returns
         -------
-        single value or a tuple
+        self : dict
         """
-        if isinstance(config, Config):
-            value = config.pop(variables, None, **kwargs)
+        if isinstance(config, dict):
+            items = config.items()
+        elif isinstance(config, (tuple, list)):
+            items = config
+            if any([not isinstance(item, (tuple, list)) for item in items]):
+                raise ValueError('Tuple or list should contain only tuples or lists')
+            if any([len(item) != 2 for item in items]):
+                raise ValueError('Tuples in list should represent pairs key-value'
+                                 ', and therefore must be always the length of 2')
+
+        for key, value in items:
+            if isinstance(key, str):
+                key = '/'.join(filter(None, key.split('/')))
+            self.put(key, value)
+
+        return self
+
+    def put(self, key, value):
+        """ Put a new key into config recursively.
+
+        Parameters
+        ----------
+        key : hashable object
+            key to add. '/' is used to put value into nested dict
+        value : masc
+        """
+        if isinstance(key, str) and '/' in key:
+            config = self
+            parent, child = key.split('/', 1)
+
+            if parent in config and isinstance(config[parent], Config): # for example, we put value=3 with key='a/c' into the
+                config[parent].update(Config({child: value}))           # config = {'a': {'b': 1}} and want to receive {'a': {'b': 1, 'c': 3}}
+            else:
+                config[parent] = Config({child: value})
```

Comment:
We will silently rewrite items like `{'a': 3}` by `{'a/b': 2}`. Should it be a warning or exception?

Reply:
> We will silently rewrite items like `{'a': 3}` by `{'a/b': 2}`. Should it be a warning or exception?

I think, yes, a warning would be good here